packages/impit-client/src/index.ts

```diff
@@ -20,65 +26,72 @@ export class ImpitHttpClient implements BaseHttpClient {
 
-        this.maxRedirects = options?.maxRedirects ?? 10;
-        this.followRedirects = options?.followRedirects ?? true;
-    }
+    private getClient(options: ImpitOptions) {
+        const cacheKey = JSON.stringify(options);
 
-    /**
-     * Converts the body of a `HttpRequest` to a format that can be passed to `impit`.
-     */
-    private async intoImpitBody<TResponseType extends keyof ResponseTypes>(
-        body: Exclude<HttpRequest<TResponseType>['body'], undefined>,
-    ): Promise<string | Uint8Array> {
-        if (typeof body === 'string' || isTypedArray(body)) {
-            return body;
+        if (this.clientCache.get(cacheKey)) {
+            return this.clientCache.get(cacheKey)!;
         }
 
-        if (body instanceof ReadableStream) {
-            const reader = body.getReader();
-            const buffer = new Uint8Array();
-
-            while (true) {
-                const { done, value } = await reader.read();
+        const client = new Impit(options);
+        this.clientCache.add(cacheKey, client);
 
-                if (done) return buffer;
+        return client;
+    }
 
-                buffer.set(value, buffer.length);
-            }
-        }
+    constructor(options?: Omit<ImpitOptions, 'proxyUrl'> & { maxRedirects?: number }) {
+        this.impitOptions = options ?? {};
 
-        throw new Error('Unsupported body type.');
+        this.maxRedirects = options?.maxRedirects ?? 10;
+        this.followRedirects = options?.followRedirects ?? true;
     }
 
     /**
      * Flattens the headers of a `HttpRequest` to a format that can be passed to `impit`.
      * @param headers `SimpleHeaders` object
      * @returns `Record<string, string>` object
      */
-    private flattenHeaders<TResponseType extends keyof ResponseTypes>(
-        headers: Exclude<HttpRequest<TResponseType>['headers'], undefined>,
-    ): Record<string, string> {
-        const result: Record<string, string> = {};
+    private intoHeaders<TResponseType extends keyof ResponseTypes>(
+        headers?: Exclude<HttpRequest<TResponseType>['headers'], undefined>,
+    ): Headers | undefined {
+        if (!headers) {
+            return undefined;
+        }
+
+        const result = new Headers();
 
         for (const headerName of Object.keys(headers)) {
             const headerValue = headers[headerName];
 
-            if (headerValue === undefined) continue;
+            for (const value of Array.isArray(headerValue) ? headerValue : [headerValue]) {
+                if (value === undefined) continue;
 
-            if (Array.isArray(headerValue)) {
-                result[headerName] = headerValue[0];
-                continue;
+                result.append(headerName, value);
             }
-
-            result[headerName] = headerValue;
         }
 
         return result;
     }
 
+    private isGeneratorObject<T>(value: any): value is Generator<T> | AsyncGenerator<T> {
```

Comment:
This just adds the type guard on top of the node util, correct?

Reply:
Yes, and as we cast everything to `any` anyway, it's not really useful :) Removing for now.